packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -565,8 +574,18 @@ export class StripeWebhookHandler extends StripeHandler {
     let account;
     try {
       // If the user's account has not been deleted, we should have already
-      // sent email at subscription update when cancel_at_period_end = true
-      await this.db.account(uid);
+      // sent email at subscription update when cancel_at_period_end = true,
+      // _or_ the cancellation is from Stripe due to failed retries or the
+      // PayPal processor, which we'll handle here.
+      account = await this.db.account(uid);
+      await this.mailer.sendSubscriptionFailedPaymentsCancellationEmail(
+        account.emails,
+        account,
+        {
+          acceptLanguage: account.locale,
+          ...invoiceDetails,
+        }
+      );
```

Comment:
According to my understanding of [Stripe's docs on subscription cancellations](https://stripe.com/docs/billing/subscriptions/cancel#events), even users who voluntarily cancel a subscription (i.e. we mark it as `cancel_at_period_end` and get a `subscription.updated` webhook) will still trigger a `subscription.deleted` webhook once the subscription period ends. With the logic as written, we would send users who voluntarily cancel this involuntary cancellation email at that time as well.

I think we will want to update this logic to check for a voluntary cancellation prior. I'm not sure what the best way is to do that -- can we look at the webhook event history for this `subscription` and find the most recent `subscription.updated` event for the current billing period to see if the change was to set `cancel_at_period_end`?

Reply:
This was my biggest concern while working on this patch.  The diff view is not great for this...if you look at line 568, you see that we return early if it was a voluntary cancellation.  I'll see about adding a test for it.

Reply:
Oh okay I see that; my bad... So the logic looks right to me in that case for voluntary cancellations through Subscription Management. If Support cancels a subscription for a customer (using `cancel_at_period_end`), do they modify this metadata property? I know that's an edge case...

Reply:
I won't block on adding a test for this, since this isn't new logic that you added as part of this PR.

Reply:
> If Support cancels a subscription for a customer (using cancel_at_period_end), do they modify this metadata property?

hmm good point.  I'm guessing that they don't.

Reply:
Is it worth pinging JR in the Jira ticket for this bug to ask them to update their instructions to include modifying this?

Reply:
Perhaps as a last resort.  I prefer to not rely on a manual process if we can avoid it.